runtime/j9vm/javanextvmi.cpp

```diff
@@ -444,10 +452,47 @@ JVM_VirtualThreadUnmountEnd(JNIEnv *env, jobject thread, jboolean lastUnmount)
 
 	vmFuncs->internalExitVMToJNI(currentThread);
 
-	Trc_SC_VirtualThreadUnmountEnd_Exit(currentThread, thread, lastUnmount);
+	Trc_SC_VirtualThreadUnmountEnd_Exit(currentThread, thread);
 }
 #endif /* JAVA_SPEC_VERSION >= 19 */
 
+#if (JAVA_SPEC_VERSION >= 19) && (JAVA_SPEC_VERSION < 21)
+JNIEXPORT void JNICALL
+JVM_VirtualThreadMountBegin(JNIEnv *env, jobject thread, jboolean firstMount)
+{
+	virtualThreadMountBegin(env, thread);
+}
+
+JNIEXPORT void JNICALL
+JVM_VirtualThreadMountEnd(JNIEnv *env, jobject thread, jboolean firstMount)
+{
+	virtualThreadMountEnd(env, thread);
+	if (firstMount) {
+		J9VMThread *currentThread = (J9VMThread *)env;
+		J9JavaVM *vm = currentThread->javaVM;
+		TRIGGER_J9HOOK_VM_VIRTUAL_THREAD_STARTED(vm->hookInterface, currentThread);
+	}
+}
+
+JNIEXPORT void JNICALL
+JVM_VirtualThreadUnmountBegin(JNIEnv *env, jobject thread, jboolean lastUnmount)
+{
+	if (lastUnmount) {
+		J9VMThread *currentThread = (J9VMThread *)env;
+		J9JavaVM *vm = currentThread->javaVM;
+		TRIGGER_J9HOOK_VM_VIRTUAL_THREAD_END(vm->hookInterface, currentThread);
+		unsetParentVthread((J9VMThread *)env, thread);
+	}
+	virtualThreadUnmountBegin(env, thread);
+}
+
+JNIEXPORT void JNICALL
+JVM_VirtualThreadUnmountEnd(JNIEnv *env, jobject thread, jboolean lastUnmount)
+{
+	virtualThreadUnmountEnd(env, thread);
```

Comment:
Why does this ignore `lastUnmount`?

Reply:
In our implementation, `lastUnmount` is handled completely in `JVM_VirtualThreadUnmountBegin`. So, it does not get used here. The same response applies to the question for `firstMount`: https://github.com/eclipse-openj9/openj9/pull/17125/files#r1220069964.